server/service/src/sync/translations/special/name_merge.rs

```diff
@@ -56,7 +60,57 @@ impl SyncTranslation for NameMergeTranslation {
             })
             .collect();
 
-        Ok(Some(IntegrationRecords::from_upserts(upsert_records)))
+        let name_store_join_repo = NameStoreJoinRepository::new(connection);
+        let name_store_joins_for_delete = name_store_join_repo.query(Some(
+            NameStoreJoinFilter::new().name_id(EqualFilter::equal_to(&data.merge_id_to_delete)),
+        ))?;
+        let name_store_joins_for_keep = name_store_join_repo.query(Some(
+            NameStoreJoinFilter::new().name_id(EqualFilter::equal_to(&data.merge_id_to_keep)),
+        ))?;
+        let mut deletes: Vec<PullDeleteRecord> = vec![];
+
+        // We need to delete the name_store_joins that are no longer needed after the merge
+        // Situation A: ("Joined to" meaning store->nsj->name_link->name)
+        // storeA joined to nameK
+        // storeA joined to nameD
+        // storeB joined to nameD
+        // nameD merged into nameK
+        // storeA joined to nameK
+        // storeA joined to nameK (delete this join to avoid showing twice in lists seemingly as a duplicate)
+        // storeB joined to nameK (make sure we don't accidentally delete this one, or visibility of nameK will be lost for storeB)
+        name_store_joins_for_delete.iter().for_each(|nsj_delete| {
+            if name_store_joins_for_keep.iter().any(|nsj_keep| {
+                nsj_keep.name_store_join.store_id == nsj_delete.name_store_join.store_id
+            }) {
+                deletes.push(PullDeleteRecord {
+                    id: nsj_delete.name_store_join.id.clone(),
+                    table: PullDeleteRecordTable::NameStoreJoin,
+                });
+            }
+        });
+
+        // Situation B:
+        // storeK.name_id == nameK.id
+        // storeK joined to nameD
+        // nameD merged into nameK
+        // storeK joined to nameK (delete the join before this happens, stores shouldn't be visible to themselves)
+        let store_repo = StoreRepository::new(connection);
+        let store_filter =
+            StoreFilter::new().name_id(EqualFilter::equal_to(&data.merge_id_to_keep));
+        let store = store_repo.query_one(store_filter)?;
```

Comment:
This is only being used here... can it be 
```
let store = StoreRepository::new(connection).query_one(StoreFilter::new().name_id(EqualFilter::equal_to(&data.merge_id_to_keep));
```
?

Reply:
It was hang over from when I did several queries and I thought it was more readable so kept it - though I'm inconsistent in the PR elsewhere.

Actually the formatter handles it nicely haha. f0c2bade134c